### PR TITLE
fix: react-combobox add listbox max-height and shift shadow style location

### PR DESCRIPTION
**Pull request by smhigley**
## Changes

- The listbox popup did not have a max-height, so grew with its contents past the screen height. Now it's set to `80vh`
- box-shadow and border-radius were defined in `useListboxStyles`. Those have been moved to `useComboboxStyles` and `useDropdownStyles` so Listbox can be more easily re-used in different contexts (and because the shadow and border-radius are more tied to the popup, not the listbox)

Max-height is part of #26069
Shadow styles are part of #25866

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                             |     Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                                   Change |
| :---------------------------------------------------------------------------------------------------------------------------- | ---------------------------: | ---------------------------: | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr> | `80.935 kB`<br />`25.855 kB` | `80.722 kB`<br />`25.872 kB` | `-213 B` <br />`17 B`  |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/ComboboxField.fixture.js'>ComboboxField</abbr>                    | `78.821 kB`<br />`25.999 kB` | `78.608 kB`<br />`26.021 kB` | `-213 B` <br />`22 B`  |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr> | `79.721 kB`<br />`25.651 kB` | `79.508 kB`<br />`25.659 kB` |  `-213 B` <br />`8 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |         Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              | `59.888 kB`<br />`16.652 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `187.586 kB`<br />`52.65 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            | `34.379 kB`<br />`11.322 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |   `6.069 kB`<br />`2.053 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/3079711d3af954e45081347e7a5abb963ef78abf'>3079711d3af954e45081347e7a5abb963ef78abf</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26470/perf-test-react-components/Avatar-mount_base.html">1320</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26470/perf-test-react-components/Avatar-mount.html">1301</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26470/perf-test-react-components/Button-mount_base.html">939</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26470/perf-test-react-components/Button-mount.html">949</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26470/perf-test-react-components/FluentProvider-mount_base.html">1591</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26470/perf-test-react-components/FluentProvider-mount.html">1601</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26470/perf-test-react-components/FluentProviderWithTheme-mount_base.html">636</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26470/perf-test-react-components/FluentProviderWithTheme-mount.html">618</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26470/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">605</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26470/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">581</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26470/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">630</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26470/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">627</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26470/perf-test-react-components/InfoButton-mount_base.html">565</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26470/perf-test-react-components/InfoButton-mount.html">572</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26470/perf-test-react-components/MakeStyles-mount_base.html">1912</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26470/perf-test-react-components/MakeStyles-mount.html">1905</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26470/perf-test-react-components/Persona-mount_base.html">2998</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26470/perf-test-react-components/Persona-mount.html">2993</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26470/perf-test-react-components/SpinButton-mount_base.html">2496</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26470/perf-test-react-components/SpinButton-mount.html">2543</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: change/@fluentui-react-combobox-180e14ee-bb08-41bc-ae39-3046c6fa0c5e.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "fix: add popup maxHeight, and move shadow styles to combobox/dropdown\"",
```
Comment (khmakoto) [inline change/@fluentui-react-combobox-180e14ee-bb08-41bc-ae39-3046c6fa0c5e.json:3]:
```suggestion
  "comment": "fix: add popup maxHeight, and move shadow styles to combobox/dropdown",
```

**Review by khmakoto — APPROVED** [review]

**Pull request merged**